core/src/main/scala/org/apache/spark/ui/SparkUI.scala

```diff
@@ -126,6 +126,10 @@ private[spark] class SparkUI private (
       ))
     ))
   }
+
+  def getApplicationInfo(appId: String): Option[ApplicationInfo] = {
+    throw new UnsupportedOperationException()
```

Comment:
I think this would be better to return the expected result (the info if the id matches or a None if it doesn't) rather than throw an UnsupportedException, just for consistency